cli/src/main/java/com/devonfw/tools/ide/process/ProcessResultImpl.java

```diff
@@ -55,4 +58,25 @@ public List<String> getErr() {
     return this.err;
   }
 
+  @Override
+  public void log(IdeLogLevel level, IdeContext context) {
+
+    if (this.out != null && !this.out.isEmpty() && level == IdeLogLevel.INFO) {
+      doLog(level, this.out, context);
+    }
+    if (this.err != null && !this.err.isEmpty() && level == IdeLogLevel.ERROR) {
+      doLog(level, this.err, context);
+    }
```

Comment:
You are not implementing what you specified in the JavaDoc.
Why should you ignore `out` and `err` if the passed log-level is not as expected/hardcoded here?
This code is really confusing me...

Reply:
Optimized and removed check for log level now.